packages/ui/src/components/Counter/Counter.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useState } from 'react';
+import { useInView } from 'react-intersection-observer';
+import { animate, useMotionValue } from 'framer-motion';
+
+import { Typography, TypographyProps } from '../Typography';
+
+interface AnimatedCounterProps extends Omit<TypographyProps, 'children'> {
+  valueFrom?: number;
+  valueTo: string;
+  totalDuration?: number;
+}
+
+function digitsInString(value: string) {
+  return Number(value.match(/\d+/g)?.join('')) || 0;
+}
+
+function postfixInString(value: string) {
+  return String(value.match(/\D+/) || '');
+}
+
+function addThousandsSeparator(value: string) {
+  return value
+    .split('')
+    .reverse()
+    .join('')
+    .match(/.{1,3}/g)
+    ?.map((val) => val.split('').reverse().join(''))
+    .reverse()
+    .join(' ');
+}
```

Comment:
💭 The animation looks great and I leave approve 👍  You can change the regular functions to the arrow functions that we basically use throughout the project.

Reply:
+1 for arrow functions